### ISSUE TITLE
Could not build wheels which is required to install pyproject.toml-based projects 

### ISSUE THREAD

**Issue by 0xD4rky** [open-mmlab/mmsegmentation#3695]
Thanks for your error report and we appreciate it a lot.

**Checklist**

1. I have searched related issues but cannot get the expected help.
2. The bug has not been fixed in the latest version.

**Describe the bug**

I tried to install mmcv (2.0.0) with the follow command:
pip install mmcv==2.0.0rc4
But it fails and report an error:
"Could not build wheels for mmcv, which is required to install pyproject.toml-based projects"

My system: Windows 11

**Reproduction**

1. What command or script did you run?

 !rm -rf mmsegmentation
!git clone -b main https://github.com/open-mmlab/mmsegmentation.git 
%cd mmsegmentation
!pip install -e .
pip install mmcv==2.0.0rc4

2. Did you make any modifications on the code or config? Did you understand what you have modified?
No

4. What dataset did you use?

**Environment**

1. Please run `python mmseg/utils/collect_env.py` to collect necessary environment information and paste it here.
2. You may add addition that may be helpful for locating the problem, such as
   - How you installed PyTorch \[e.g., pip, conda, source\]
   - Other environment variables that may be related (such as `$PATH`, `$LD_LIBRARY_PATH`, `$PYTHONPATH`, etc.)

**Error traceback**

If applicable, paste the error trackback here.
Collecting mmcv==2.0.0rc4
  Downloading mmcv-2.0.0rc4.tar.gz (436 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 436.4/436.4 kB 3.0 MB/s eta 0:00:00
  Preparing metadata (setup.py) ... done
Requirement already satisfied: addict in /usr/local/lib/python3.10/dist-packages (from mmcv==2.0.0rc4) (2.4.0)
Requirement already satisfied: mmengine>=0.2.0 in /usr/local/lib/python3.10/dist-packages (from mmcv==2.0.0rc4) (0.10.4)
Requirement already satisfied: numpy in /usr/local/lib/python3.10/dist-packages (from mmcv==2.0.0rc4) (1.25.2)
Requirement already satisfied: packaging in /usr/local/lib/python3.10/dist-packages (from mmcv==2.0.0rc4) (24.0)
Requirement already satisfied: Pillow in /usr/local/lib/python3.10/dist-packages (from mmcv==2.0.0rc4) (9.4.0)
Requirement already satisfied: pyyaml in /usr/local/lib/python3.10/dist-packages (from mmcv==2.0.0rc4) (6.0.1)
Requirement already satisfied: yapf in /usr/local/lib/python3.10/dist-packages (from mmcv==2.0.0rc4) (0.40.2)
Requirement already satisfied: matplotlib in /usr/local/lib/python3.10/dist-packages (from mmengine>=0.2.0->mmcv==2.0.0rc4) (3.7.1)
Requirement already satisfied: rich in /usr/local/lib/python3.10/dist-packages (from mmengine>=0.2.0->mmcv==2.0.0rc4) (13.4.2)
Requirement already satisfied: termcolor in /usr/local/lib/python3.10/dist-packages (from mmengine>=0.2.0->mmcv==2.0.0rc4) (2.4.0)
Requirement already satisfied: opencv-python>=3 in /usr/local/lib/python3.10/dist-packages (from mmengine>=0.2.0->mmcv==2.0.0rc4) (4.8.0.76)
Requirement already satisfied: importlib-metadata>=6.6.0 in /usr/local/lib/python3.10/dist-packages (from yapf->mmcv==2.0.0rc4) (7.1.0)
Requirement already satisfied: platformdirs>=3.5.1 in /usr/local/lib/python3.10/dist-packages (from yapf->mmcv==2.0.0rc4) (4.2.2)
Requirement already satisfied: tomli>=2.0.1 in /usr/local/lib/python3.10/dist-packages (from yapf->mmcv==2.0.0rc4) (2.0.1)
Requirement already satisfied: zipp>=0.5 in /usr/local/lib/python3.10/dist-packages (from importlib-metadata>=6.6.0->yapf->mmcv==2.0.0rc4) (3.19.0)
Requirement already satisfied: contourpy>=1.0.1 in /usr/local/lib/python3.10/dist-packages (from matplotlib->mmengine>=0.2.0->mmcv==2.0.0rc4) (1.2.1)
Requirement already satisfied: cycler>=0.10 in /usr/local/lib/python3.10/dist-packages (from matplotlib->mmengine>=0.2.0->mmcv==2.0.0rc4) (0.12.1)
Requirement already satisfied: fonttools>=4.22.0 in /usr/local/lib/python3.10/dist-packages (from matplotlib->mmengine>=0.2.0->mmcv==2.0.0rc4) (4.52.4)
Requirement already satisfied: kiwisolver>=1.0.1 in /usr/local/lib/python3.10/dist-packages (from matplotlib->mmengine>=0.2.0->mmcv==2.0.0rc4) (1.4.5)
Requirement already satisfied: pyparsing>=2.3.1 in /usr/local/lib/python3.10/dist-packages (from matplotlib->mmengine>=0.2.0->mmcv==2.0.0rc4) (3.1.2)
Requirement already satisfied: python-dateutil>=2.7 in /usr/local/lib/python3.10/dist-packages (from matplotlib->mmengine>=0.2.0->mmcv==2.0.0rc4) (2.8.2)
Requirement already satisfied: markdown-it-py>=2.2.0 in /usr/local/lib/python3.10/dist-packages (from rich->mmengine>=0.2.0->mmcv==2.0.0rc4) (3.0.0)
Requirement already satisfied: pygments<3.0.0,>=2.13.0 in /usr/local/lib/python3.10/dist-packages (from rich->mmengine>=0.2.0->mmcv==2.0.0rc4) (2.16.1)
Requirement already satisfied: mdurl~=0.1 in /usr/local/lib/python3.10/dist-packages (from markdown-it-py>=2.2.0->rich->mmengine>=0.2.0->mmcv==2.0.0rc4) (0.1.2)
Requirement already satisfied: six>=1.5 in /usr/local/lib/python3.10/dist-packages (from python-dateutil>=2.7->matplotlib->mmengine>=0.2.0->mmcv==2.0.0rc4) (1.16.0)
Building wheels for collected packages: mmcv
  error: subprocess-exited-with-error
  
  × python setup.py bdist_wheel did not run successfully.
  │ exit code: 1
  ╰─> See above for output.
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
  Building wheel for mmcv (setup.py) ... error
  ERROR: Failed building wheel for mmcv
  Running setup.py clean for mmcv
Failed to build mmcv
ERROR: Could not build wheels for mmcv, which is required to install pyproject.toml-based projects

```none
A placeholder for trackback.
```

**Bug fix**

If you have already identified the reason, you can provide the information here. If you are willing to create a PR to fix it, please also leave a comment here and that would be much appreciated!

**Comment by yyxxiiaaoo** [comment]
I have the same issue,have you solved it?

**Comment by 0xD4rky** [comment]
yes I have done it

**Comment by 0xD4rky** [comment]
> I have the same issue,have you solved it?

Yes I have resolved the issue. It was mainly related to the incompatibility of torch and torchvision in the newer updates for the same.

**Comment by yyxxiiaaoo** [comment]
thanks
